src/modules/dashboard/components/Dialogs/EscrowFundsDialog/EscrowFundsDialog.tsx

```diff
@@ -0,0 +1,309 @@
+    collective approval first. To do so create a motion. 
+    It is like asking your teammates if they agree to fund your expenditure.
+    `,
+  },
+  allocationHeader: {
+    id: 'dashboard.EscrowFundsDialog.allocationHeader',
+    defaultMessage: 'Allocation and resources',
+  },
+  allocationTeam: {
+    id: 'dashboard.EscrowFundsDialog.allocationTeam',
+    defaultMessage: 'Team',
+  },
+  allocationBalance: {
+    id: 'dashboard.EscrowFundsDialog.allocationTeam',
+    defaultMessage: 'Balance',
+  },
+  fundsHeader: {
+    id: 'dashboard.EscrowFundsDialog.fundsHeader',
+    defaultMessage: 'Required funds',
+  },
+  textareaLabel: {
+    id: 'dashboard.EscrowFundsDialog.textareaLabel',
+    defaultMessage: `Explain why you're funding this expenditure (optional)`,
+  },
+  confirmText: {
+    id: 'dashboard.EscrowFundsDialog.confirmText',
+    defaultMessage: 'Create motion',
+  },
+});
+
+const displayName = 'dashboard.EscrowFundsDialog';
+
+interface FormValues {
+  force: boolean;
+  filteredDomainId: string;
+  annotationMessage?: string;
+}
+
+interface Props {
+  cancel: VoidFunction;
+  close: VoidFunction;
+  onClick: VoidFunction;
+  /*
+   * To retrieve teams and tokens
+   */
+  colonyName: string;
+  isVotingExtensionEnabled: boolean;
+}
+
+const EscrowFundsDialog = ({
+  cancel,
+  onClick,
+  close,
+  colonyName,
+  isVotingExtensionEnabled,
+}: Props) => {
+  /*
+   * Hook to distinct the logic from render
+   */
+  const {
+    colonyData,
+    renderActiveOption,
+    filterDomains,
+    balanceOptions,
+    requiredFunds,
+    handleMotionDomainChange,
+    domainID,
+    loading,
+    renderActiveDomainOption,
+  } = useEscrowFundsDialog(colonyName);
+  const [isForce, setIsForce] = useState(false);
+
+  const handleSubmit = useCallback(() => {
+    onClick();
+    close();
+  }, [onClick, close]);
+
+  const getFormAction = useCallback(
+    (actionType: 'SUBMIT' | 'ERROR' | 'SUCCESS') => {
+      const actionEnd = actionType === 'SUBMIT' ? '' : `_${actionType}`;
+
+      // @TODO these action types are for mocking purposes, change to correct ones
+      return isVotingExtensionEnabled && !isForce
+        ? ActionTypes[`COLONY_ACTION_GENERIC${actionEnd}`]
+        : ActionTypes[`COLONY_ACTION_GENERIC${actionEnd}`];
+    },
+    [isVotingExtensionEnabled, isForce],
+  );
+
+  return (
+    <Dialog cancel={cancel}>
+      <ActionForm
+        submit={getFormAction('SUBMIT')}
+        error={getFormAction('ERROR')}
+        success={getFormAction('SUCCESS')}
+        initialValues={{ force: false }}
+        onSubmit={handleSubmit}
+      >
+        {(formValues: FormikProps<FormValues>) => {
+          if (formValues.values.force !== isForce) {
+            setIsForce(formValues.values.force);
+          }
+          return (
+            <>
+              <div className={styles.dialogContainer}>
+                <DialogSection appearance={{ theme: 'heading' }}>
+                  {loading ? (
```

Comment:
I wanted to check the loading logic and just assigned true to loading. The app crashed and gave an error. This shouldn't happen. Please, fix.
<img width="1038" alt="Screenshot 2022-07-19 at 15 16 59" src="https://user-images.githubusercontent.com/34057551/179748322-1266e4fa-4ec8-41cb-931e-2eea24a861be.png">

Reply:
To be honest, I don't really see a point of using loading for both cases in this code block... Maybe I'm missing something but in places where we use `DomainDropdown` we never use loading for this part (you can check dialogs with DomainDropdown as an example)
To not wait for the colonyFromName query result - you can just call query in a parent component and pass `colony` instead of `colonyName` prop.
Up to you to decide as you are more versed in the code but if you use loading the error has to be fixed.